test/functional/terminal/tui_spec.lua

```diff
@@ -173,6 +177,79 @@ describe('tui', function()
+      {1: }                                                 |
+      ~                                                 |
+      ~                                                 |
+      ~                                                 |
+      [No Name]                                         |
+      gained                                            |
+      -- TERMINAL --                                    |
+    ]])
+    feed('\x1b[O')
+    screen:expect([[
+      {1: }                                                 |
+      ~                                                 |
+      ~                                                 |
+      ~                                                 |
+      [No Name]                                         |
+      lost                                              |
+      -- TERMINAL --                                    |
+    ]])
+
+    -- In command-line mode
+    feed('\x1b')
+    feed(':')
+    feed('\x1b[I')
+    screen:expect([[
+                                                        |
+      ~                                                 |
+      ~                                                 |
+      ~                                                 |
+      [No Name]                                         |
+      g{1:a}ined                                            |
+      -- TERMINAL --                                    |
+    ]])
+    feed('\x1b[O')
+    screen:expect([[
+                                                        |
+      ~                                                 |
+      ~                                                 |
+      ~                                                 |
+      [No Name]                                         |
+      l{1:o}st                                              |
+      -- TERMINAL --                                    |
+    ]])
+
+    -- In terminal mode
+    execute('set shell='..nvim_dir..'/shell-test')
+    execute('set laststatus=0')
+    feed('\x1b')
+    execute('terminal')
+    feed('\x1b[I')
+    screen:expect([[
+      ready $                                           |
+      [Process exited 0]{1: }                               |
+                                                        |
+                                                        |
+                                                        |
+      gained                                            |
+      -- TERMINAL --                                    |
+    ]])
+   feed('\x1b[O')
+    screen:expect([[
+      ready $                                           |
+      [Process exited 0]{1: }                               |
+                                                        |
+                                                        |
+                                                        |
+      lost                                              |
+      -- TERMINAL --                                    |
+    ]])
```

Comment:
Since these tests are all related to perhaps these should be refactored into individual `it()` tests inside a `describe()` block?